### PR TITLE
QUA-431: update sonarlint core to 4.6.0.2652

### PR DESCRIPTION
**Pull request by dantevvp**
This PR updates the wrapper dependencies to support SonarJava version 6.12

Packages updated:

Package | from | to
---------|------|---
org.sonarsource.sonarlint.core:sonarlint-core|2.17.0.899|4.6.0.2652
org.sonarsource.sonarlint.core:sonarlint-api|2.17.0.899|4.6.0.2652
org.sonarsource.java:sonar-java-plugin|4.14.0.11784|6.12.0.24852
com.google.code.findbugs:jsr305| --- | 3.0.2

## Version updates

- Update sonarlint core to version 4.6.0.2652
- Update sonarlint api to version 4.6.0.2652
- Update sonar java plugin to version 6.12.0.24852
- Add com.google.code.findbugs (needed for overriding the `InputFileFinder` class in `sonarlint-cli`)

### InputFileFinder

- Override `sonarlint-cli`'s `InputFileFinder` to implement new `uri` and `relativePath` methods in `DefaultClientInputFile`, [not present in its latest version](https://github.com/SonarSource/sonarlint-cli/blob/595cebc7183f4456a62797743a2ea4738ccddcd8/src/main/java/org/sonarlint/cli/InputFileFinder.java#L127).

### StandaloneSonarLint

- Update `StandaloneAnalysisConfiguration` to be built using its [new `Builder` subclass](https://github.com/SonarSource/sonarlint-core/blob/4.12.0.18542/client-api/src/main/java/org/sonarsource/sonarlint/core/client/api/standalone/StandaloneAnalysisConfiguration.java).
- Override `getRuleDetails` method to call `get()` on the [new returned type `Optional<RuleDetails>` instead of `RuleDetails`](https://github.com/SonarSource/sonarlint-core/blob/4.12.0.18542/client-api/src/main/java/org/sonarsource/sonarlint/core/client/api/standalone/StandaloneSonarLintEngine.java).

### CategoriesTest

- Change deprecated rule keys [`S00115`](https://github.com/SonarSource/sonar-java/blob/6.12.0.24852/java-checks/src/main/java/org/sonar/java/checks/naming/BadConstantNameCheck.java) and [`S00122`](https://github.com/SonarSource/sonar-java/blob/6.12.0.24852/java-checks/src/main/java/org/sonar/java/checks/TooManyStatementsPerLineCheck.java) to `S115` and `S122`

### Factory

- Change deprecated rule repository [`squid` to `java`](https://github.com/SonarSource/sonar-java/pull/2729)

### CODE REVIEW

**Comment by dantevvp** [comment]
I've also opened a new branch `channel-sonar-6-12` to eventually pull from in the `codeclimate/codeclimate-sonar-java` repo without overwriting the wrapper's `channel-sonar-5-14` image.

**Comment by dantevvp** [comment]
@fede-moya I added a test for a new rule that passes correctly, indicating that new rules are being installed. This would mean that SonarJava 6.12 is working correctly in the wrapper.

However, new rules don't seem to fire new issues, even when setting `minimum_severity` to `info` and `sonar.java.source` to `15`. I also tried enabling the check manually in the `checks` attribute, but had no luck.

To be clear, issues do appear, but only those issues from older sonar-java versions. I've tried testing issues from newer versions and haven't been able to see them appearing when they should.

I haven't found any code/task that updates the rules repository. I believe this is done by just installing the new `sonar-java-plugin` version.

File: src/test/java/support/Factory.java [hunk 1]
```diff
@@ -14,6 +15,6 @@ public static StandaloneSonarLintEngineImpl sonarlint() {
     }
 
     public static RuleDetails createRule(String key) {
-        return sonarlint().getRuleDetails("squid:" + key);
+        return sonarlint().getRuleDetails("java:" + key).get();
     }
-}
```
Comment (f-moya) [inline src/test/java/support/Factory.java:19]: 👍🏼

File: src/test/java/support/Factory.java [hunk 2]
```diff
@@ -1,5 +1,6 @@
 package support;
 
+import java.util.Collection;
```
Comment (f-moya) [inline src/test/java/support/Factory.java]: is it really needed ? doesn't seem to be used

File: src/test/java/cc/models/CategoriesTest.java [hunk 3]
```diff
@@ -78,8 +78,13 @@ public void set_bug_risk_for_suspicious_code_smells() throws Exception {
         assertThat(getCategoriesForRule("S1186")).contains(Categories.Category.BUG_RISK);
     }
 
+    @Test
+    public void set_clarity_for_single_line_text_blocks() throws Exception {
+        assertThat(getCategoriesForRule("S5663")).contains(Categories.Category.CLARITY);
+    }
```
Comment (f-moya) [inline src/test/java/cc/models/CategoriesTest.java:84]: 👍🏼

**Review by f-moya — APPROVED** [review]
Left some minor comments, this work was previously review IRL with Dante.

**Comment by dantevvp** [comment]
I'm leaving the base branch as `channel-sonar-6-12` for now. Changing the base branch to `beta` adds lots of commits and changed files because it's a very old branch. It might even be better to just rename this `channel-sonar-6-12` to beta and remove the beta branch altogether

**Comment by f-moya** [comment]
@dantevvp just updated [beta](https://github.com/codeclimate/codeclimate-ss-analyzer-wrapper/tree/beta), feel free to update the base to the now updated [beta](https://github.com/codeclimate/codeclimate-ss-analyzer-wrapper/tree/beta) 😎

**Pull request merged**